apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs

```diff
@@ -30,23 +29,30 @@ namespace Google.Cloud.Storage.V1.Tests;
 /// </summary>
 public class RetryTest
 {
-    /// <summary>
-    /// A set of retry options using default timing, but a predicate that retries on 404 (NotFound) only.
-    /// </summary>
-    private static readonly RetryOptions s_customPredicateOptions = new RetryOptions(RetryTiming.Default, RetryPredicate.FromErrorCodes(404));
+    private static readonly RetryTiming s_customRetryTiming = new(TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(5), 2);
+    private static readonly RetryPredicate s_customPredicate = RetryPredicate.FromErrorCodes(404, 400);
+    private static readonly RetryOptions s_customRetryOptions = new(s_customRetryTiming, s_customPredicate);
 
     public static TheoryData<TestCase> TestCases { get; } = new TheoryData<TestCase>
     {
-        new TestCase("SimpleSuccess", RetryOptions.Never, new[] { OK }, new[] { 0 }, true),
-        new TestCase("ImmediateFailure", RetryOptions.IdempotentRetryOptions, new[] { BadRequest }, new[] { 0 }, false),
-        new TestCase("SingleRetry", RetryOptions.IdempotentRetryOptions, new[] { InternalServerError, OK }, new[] { 0, 1 }, true),
-        new TestCase("DoubleRetry", RetryOptions.IdempotentRetryOptions, new[] { InternalServerError, BadGateway, OK }, new[] { 0, 1, 3 }, true),
-        // We default to a maximum of 3 requests
+        new TestCase("SimpleSuccess_NoRetry", RetryOptions.IdempotentRetryOptions, new[] { OK }, new[] { 0 }, true),
+        new TestCase("SuccessAfterMaxRetry", RetryOptions.IdempotentRetryOptions, new[] { BadGateway, BadGateway, OK }, new[] { 0, 1, 3 }, true),
+        new TestCase("FailureAfterMaxRetry", RetryOptions.IdempotentRetryOptions, new[] { BadGateway, BadGateway, BadGateway }, new[] { 0, 1, 3 }, false),
+
+        new TestCase("CustomTiming_SuccessAfterRetry",  s_customRetryOptions, new[] { NotFound, BadRequest, OK }, new[] { 0, 1, 3 }, true),
+        new TestCase("CustomTiming_FailureAfterRetry",  s_customRetryOptions, new[] { NotFound, BadRequest, BadRequest }, new[] { 0, 1, 3}, false),
+
+        new TestCase("CustomPredicate_Success", s_customRetryOptions, new[] { OK }, new[] { 0 }, true),
+        new TestCase("CustomPredicate_ImmediateFailure", s_customRetryOptions, new[] { InternalServerError }, new[] { 0 }, false),
+        new TestCase("CustomPredicate_SingleRetry", s_customRetryOptions, new[] { NotFound, OK }, new[] { 0, 1 }, true),
+        new TestCase("CustomPredicate_DoubleRetry",  s_customRetryOptions, new[] { NotFound, BadRequest, OK }, new[] { 0, 1, 3 }, true),
+        new TestCase("CustomPredicate_FailureAfterRetry",  s_customRetryOptions, new[] { NotFound, BadRequest, BadRequest }, new[] { 0, 1, 3 }, false),
```

Comment:
The name here doesn't explain *why* it stops retrying - it's because it's hit the maximum number of retries.